core/src/main/java/org/elasticsearch/search/internal/InternalSearchResponse.java

```diff
@@ -19,148 +19,50 @@
 
 package org.elasticsearch.search.internal;
 
+import org.elasticsearch.action.search.SearchResponseSections;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.xcontent.ToXContent;
-import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.search.SearchHits;
-import org.elasticsearch.search.aggregations.Aggregations;
 import org.elasticsearch.search.aggregations.InternalAggregations;
-import org.elasticsearch.search.profile.ProfileShardResult;
 import org.elasticsearch.search.profile.SearchProfileShardResults;
 import org.elasticsearch.search.suggest.Suggest;
 
 import java.io.IOException;
-import java.util.Collections;
-import java.util.Map;
 
-public class InternalSearchResponse implements Streamable, ToXContent {
+/**
+ * {@link SearchResponseSections} subclass that can be serialized over the wire.
+ */
+public class InternalSearchResponse extends SearchResponseSections implements Writeable, ToXContent {
```

Comment:
Sure, thanks. I prefer this, SearchResponseSections not declaring that it implements Writeable but still providing a writeTo method.